cmd/harmony/main.go

```diff
@@ -138,11 +113,34 @@ func initSetup() {
 }
 
 func createGlobalConfig() *nodeconfig.ConfigType {
+	var err error
 	nodeConfig := nodeconfig.GetGlobalConfig()
 	nodeConfig.StakingPriKey = node.LoadStakingKeyFromFile(*stakingKeyFile, *accountIndex)
 
-	// Initialize leveldb if dbSupported.
+	nodeConfig.P2pPriKey, _, err = utils.LoadKeyFromFile(*keyFile)
+	if err != nil {
+		panic(err)
+	}
+
+	// Setup Bls keys
+	nodeConfig.BlsPriKey, nodeConfig.BlsPubKey = utils.GenKey(*ip, *port)
+	if nodeConfig.BlsPriKey == nil || nodeConfig.BlsPubKey == nil {
+		panic(fmt.Errorf("generate key error"))
+	}
+
+	// Initialize leveldb for main blockchain and beacon.
 	nodeConfig.MainDB, _ = InitLDBDatabase(*ip, *port, *freshDB, false)
+	if !*isBeacon {
+		nodeConfig.BeaconDB, _ = InitLDBDatabase(*ip, *port, *freshDB, true)
```

Comment:
Ditto here.  `BeaconDB` is a live object, not a configuration parameter.

Reply:
done.